### DISCUSSION TITLE
my app is empty while connecting to other devices

### DISCUSSION THREAD

**Discussion by PyerryQ** [zauberzeug/nicegui#1677] [Q&A]
### Question

I ran my app and accessed it on the browsers of other devices, but it's empty.What's wrong?
this is my code:
```py
from nicegui import ui, events


def upload(e:events.UploadEventArguments):
    if e.name.endswith('.txt'):
        print(e.content.read().decode('utf-8'))
    else:
        with open(e.name, 'wb') as f:
            f.write(e.content.read())
            f.close()



ui.label('upload')
ui.upload(on_upload=upload)

ui.run()
```

if I connected locally, It showed me this:
![QQ截图20230924144028](https://github.com/zauberzeug/nicegui/assets/141013159/9c76cc05-71f0-4e0d-939f-160a79a41aab)

**Comment by rodja** [comment]
I can not reproduce the problem. What is the JavaScript console logging on a different device? Whats the Network Traffic showing?

**Comment by rodja** [comment]
Maybe you have a firewall which forbids external access? Does it a simple FastAPI or Flask demo work?
